lib/database/leaderboard.php

```diff
@@ -525,25 +504,26 @@ function GetLeaderboardData($lbID, $user, $numToFetch, $offset, $friendsOnly, $n
                 $dbResult = s_mysql_query($query);
                 if ($dbResult !== false) {
                     if (mysqli_num_rows($dbResult) > 0) {
-                        //    should be 1 or 0?.. I hope...
+                        // should be 1 or 0?.. I hope...
                         $db_entry = mysqli_fetch_assoc($dbResult);
                         $db_entry['DateSubmitted'] = strtotime($db_entry['DateSubmitted']);
                         settype($db_entry['Score'], 'integer');
                         $db_entry['Rank'] = (int) $db_entry['UserRank'];
+                        // @phpstan-ignore-next-line
                         unset($db_entry['UserRank']);
+                        // @phpstan-ignore-next-line
                         $db_entry['Index'] = (int) $db_entry['UserIndex'];
+                        // @phpstan-ignore-next-line
```

Comment:
Very good question! I wasn't too sure either, didn't make much sense to me either. 

It seems to happen because the same variable name with similar keys is used above. To fix this it should be refactored into a temporary variable to only hold the desired keys without the need to `unset` anything.

I try to refrain from using static analysis or style ignores. Cleaning that up will take some more attention to the function's logic, hence why I didn't take care of it yet (this function is far too long, too).